### PR TITLE
Users can sign up, log in and log out

### PR DESCRIPTION
**Pull request by sunnyuny**
- Added user signup and login on homepage profile tab
- Modified signup and login views
- Added devise helper to format devise error messages
- Included Devise::TestHelpers
[Finishes #105537380, #105537266]

### CODE REVIEW

File: test/test_helper.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@
 class ActiveSupport::TestCase
   # Setup all fixtures in test/fixtures/*.yml for all tests in alphabetical order.
   fixtures :all
-
```
Comment (muxin) [inline test/test_helper.rb]: why deleting this line?

Comment (sunnyuny) [inline test/test_helper.rb]: I did not delete any line. just added Devise::TestHelpers to pass all test cases.

File: app/helpers/devise_helper.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# Output devise error message in format
+module DeviseHelper
+  def devise_error_messages!
```
Comment (muxin) [inline app/helpers/devise_helper.rb]: What's the meaning of "!"?

File: app/helpers/devise_helper.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    return '' if resource.errors.empty?
+    
+    messages = resource.errors.full_messages.map { |msg| content_tag(:li, msg) }.join
+    html = <<-HTML
+    <div class="alert alert-error alert-danger"> <button type="button"
+    class="close" data-dismiss="alert">x</button>
+      #{messages}
+    </div>
```
Comment (muxin) [inline app/helpers/devise_helper.rb]: Could you format this part a little bit?

File: app/helpers/devise_helper.rb (hunk 2 above)
Comment (sunnyuny) [inline app/helpers/devise_helper.rb]: It is devise defined method name..here is to override the method and format the error message

File: app/helpers/devise_helper.rb (hunk 3 above)
Comment (sunnyuny) [inline app/helpers/devise_helper.rb]: sure.

File: app/assets/stylesheets/sign_in.scss [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  padding: 10px;
+  font-size: 16px;
+}
+.form-signin .form-control:focus {
+  z-index: 2;
+}
+.form-signin input[type="email"] {
+  margin-bottom: -1px;
+  border-bottom-right-radius: 0;
+  border-bottom-left-radius: 0;
```
Comment (muxin) [inline app/assets/stylesheets/sign_in.scss]: Shall we have all input round corner? border-radius=0 gives us no round corner.

File: app/assets/stylesheets/sign_in.scss [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+}
+.form-signin input[type="password"] {
+  margin-bottom: 10px;
+  border-top-left-radius: 0;
+  border-top-right-radius: 0;
```
Comment (muxin) [inline app/assets/stylesheets/sign_in.scss]: Shall we have all input round corner? border-radius=0 gives us no round corner.

File: app/assets/stylesheets/sign_in.scss [hunk 6]
```diff
@@ -0,0 +1,40 @@
+body {
+  padding-top: 40px;
+  padding-bottom: 40px;
+  background-color: #eee;
+}
+
```
Comment (muxin) [inline app/assets/stylesheets/sign_in.scss]: Why do we have a style for body here?

File: app/helpers/devise_helper.rb (hunk 2 above)
Comment (muxin) [inline app/helpers/devise_helper.rb]: OK I don't know the meaning but I guess this is right.

File: test/test_helper.rb (hunk 1 above)
Comment (muxin) [inline test/test_helper.rb]: nvm

**Pull request closed without merging**